src/yarpdataplayer-qt/CMakeLists.txt

```diff
@@ -0,0 +1,91 @@
+# Copyright (C) 2014  iCub Facility, Istituto Italiano di Tecnologia
+# Author: Daniele E. Domenichelli <daniele.domenichelli@iit.it>
+# CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+
+if(CREATE_YARPMANAGER AND YARP_HAS_QT5)
+
+  include(YarpUseQt5)
+
+  get_property(YARP_OS_INCLUDE_DIRS TARGET YARP_OS PROPERTY INCLUDE_DIRS)
+  get_property(YARP_sig_INCLUDE_DIRS TARGET YARP_sig PROPERTY INCLUDE_DIRS)
+
+  include_directories(${YARP_OS_INCLUDE_DIRS}
+                      ${YARP_sig_INCLUDE_DIRS})
+
+  set(CMAKE_INCLUDE_CURRENT_DIR TRUE)
+
+  set(yarpdataplayer_qt_SRCS aboutdlg.cpp
+                             genericinfodlg.cpp
+                             loadingwidget.cpp
+                             main.cpp
+                             mainwindow.cpp
+                             utils.cpp
+                             worker.cpp)
+  set(yarpdataplayer_qt_HDRS aboutdlg.h
+                             genericinfodlg.h
+                             loadingwidget.h
+                             log.h
+                             mainwindow.h
+                             utils.h
+                             worker.h)
+  if(MSVC)
+    list(APPEND yarpdataplayer_HDRS include/msvc/dirent.h)
+  endif()
+  set(yarpdataplayer_qt_QRC_FILES res.qrc)
+  set(yarpdataplayer_qt_UI_FILES aboutdlg.ui
+                                 genericinfodlg.ui
+                                 loadingwidget.ui
+                                 mainwindow.ui)
+  set(yarpdataplayer_qt_THRIFT_FILES dataSetPlayer.thrift)
+  if(MSVC)
+    set(yarpdataplayer_qt_RC_FILES QtDataSetPlayer_resource.rc)
+  endif()
+
+  qt5_add_resources(yarpdataplayer_qt_QRC_GEN_SRCS ${yarpdataplayer_qt_QRC_FILES})
+  qt5_wrap_ui(yarpdataplayer_qt_UI_GEN_SRCS ${yarpdataplayer_qt_UI_FILES})
+
+  include(YarpIDL)
+  yarp_idl_to_dir(${yarpdataplayer_qt_THRIFT_FILES}
```

Comment:
@drdanz weren't you going to substitute in your new yarp_all_idl or whatever it is caused, and then not include the generated code?